src/scheduler/server_info.cpp

```diff
@@ -797,7 +801,15 @@ query_server_dyn_res(server_info *sinfo)
 			else
 				log_eventf(PBSEVENT_DEBUG2, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
 					"%s = %s (\"%s\")", conf.dynamic_res[i].command_line, res_to_str(res, RF_AVAIL), buf);
-
+			if (pid != 0) {
+				if (waitpid(pid, NULL, WNOHANG) == 0) {
```

Comment:
Do we need to kill the process here?
If we made it here, then we've already set the resource, but the script could technically still be running (if the script prints, then sleeps). 
```suggestion
				kill(-pid, SIGTERM);
				if (waitpid(pid, NULL, WNOHANG) == 0) {
```

Reply:
Yes, that's an omission.